### PR TITLE
ref(ui) Update more select controls

### PR DESCRIPTION
**Pull request by markstory**
Convert another handful of select controls to react-select v3. There are some minor styling changes made to the RoleSelectControl to align it closer to what our other select controls look like and not have the menu overlap with the adjacent teams select control.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/roleSelectControl.tsx [hunk 1]
```diff
@@ -1,58 +1,68 @@
...
+        borderBottomLeftRadius: theme.borderRadius,
+        borderBottomRightRadius: theme.borderRadius,
+      }),
+      menu: provided => ({
+        ...provided,
+        borderRadius: theme.borderRadius,
+        marginTop: space(0.5),
+        width: '350px',
+        overflow: 'hidden',
+      }),
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/roleSelectControl.tsx:60]: much nicer 😍

**Review by k-fish — COMMENTED** [review]
`RoleSelectControl` disappeared is likely why the acceptance is failing 😄

File: src/sentry/static/sentry/app/components/roleSelectControl.tsx [hunk 2]
```diff
@@ -1,58 +1,68 @@
...
-    deprecatedSelectControl
+  <SelectControl
     options={
       roles &&
-      roles.map(r => ({
-        value: r.id,
-        label: r.name,
-        disabled: disableUnallowed && !r.allowed,
-      }))
+      roles.map(
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/roleSelectControl.tsx:28]: maybe we could change it to `roles?.map` ?

**Comment by priscilawebdev** [comment]
should we keep the arrow color darker when active? with the changes now it looks always disabled

image

image

**Comment by markstory** [comment]
@priscilawebdev We could update the 'new' select control. All of the new controls have slightly smaller text and a dimmer chevron compared to the deprecated select control. I had assumed this was an intentional design decision, but I can bring it up with design folks.

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**